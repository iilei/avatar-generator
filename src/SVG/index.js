import React from 'react';
import reactDom from 'react-dom/server';
import svgx from 'svgx';


export default class SVG {
  constructor() {
    // const svg = SvgJs(window);
    // const { document } = window;
    // const draw = svg(document.documentElement);
    // draw.rect(100,100).fill('yellow').move(50,50)
  }
  toString = () => {
    const render = svgx(reactDom.renderToStaticMarkup);

    const jsx = (
      <svg width={200} height={200}>
        <circle cx={100} cy={10} r={50} bg="#bada55" />
      </svg>
    );

    const svgOutput = render(jsx);

    return (svgOutput);
  }
}
