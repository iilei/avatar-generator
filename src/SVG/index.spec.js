import SVG from '.';

describe('SVG', () => {
  it('should export properly', () => {
    expect(SVG).toBeDefined();
  });

  it('should do properly', () => {
    const instance = new SVG();
    expect(instance.toString()).toMatchSvgSnapshot();
  });
});
