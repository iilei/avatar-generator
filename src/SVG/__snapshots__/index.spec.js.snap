// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SVG should do properly 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" standalone=\\"no\\"?>
<!DOCTYPE svg PUBLIC \\"-//W3C//DTD SVG 1.1//EN\\" \\"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\\">
<svg width=\\"200\\" height=\\"200\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\"><circle cx=\\"100\\" cy=\\"100\\" r=\\"50\\"/></svg>"
`;
